src/app/api/customers/[id]/edit/route.js

```diff
@@ -0,0 +1,101 @@
+import { NextResponse } from "next/server";
+import db from "../../../../utilites/db"; 
+
+async function CreatelogEditHistory(agentId, customerId, field, oldValue, newValue) {
+    const query = `
+    INSERT INTO CustomerAuditLogs (AgentId, CustomerId, FieldChanged, OldValue, NewValue, ChangeDate) 
+             VALUES (?, ?, ?, ?, ?, NOW())
+    `;
+    const values = [agentId, customerId, field, oldValue, newValue];
+  try {
+    await db(query, values);
+  } catch (error) {
+    console.error("[DB] Error logging edit history:", error);
+    throw new Error("Failed to log edit history");
+  }
+}
+export async function fetchCustomerValues(customerId) {
+    const query = `
+    SELECT * FROM Customer where CustomerId = ?
+    `;
+    const values = [customerId];
+    try {
+      const customer = await db(query, values);
+      return customer[0];
+    } catch (error) {
+      console.error("[DB] Error fetching customer:", error);
+      throw new Error("Failed to fetch customer");
+}
+}
+
+export async function UpdateCustomerField(customerId, field, newValue) {
+    const query = `
+    UPDATE Customer SET ${field} = ? WHERE CustomerId = ?
```

Comment:
```suggestion
    const allowedFields = ["Name", "Email", "PhoneNumber", "Address"];
    if (!allowedFields.includes(field)) {
    throw new Error(`Invalid field update attempt: ${field}`);
}
    UPDATE Customer SET ${field} = ? WHERE CustomerId = ?
```
This is to prevent mysql injection